pyathena/common.py

```diff
@@ -142,24 +142,59 @@ def _build_start_query_execution_request(self, query, work_group=None, s3_stagin
             })
         return request
 
-    def _execute(self, operation, parameters=None, work_group=None, s3_staging_dir=None):
+    def _find_previous_query_id(self, query, work_group, cache_size):
+        try:
+            next_token = None
+            while cache_size > 0:
+                n = min(cache_size, 50)  # 50 is max allowed by AWS API
+                cache_size -= n
+                request = {'MaxResults': n}
+                if work_group is not None:
+                    request['WorkGroup'] = work_group
+                if next_token is not None:
+                    request['NextToken'] = next_token
+                response = retry_api_call(self.connection._client.list_query_executions,
+                                          config=self._retry_config,
+                                          logger=_logger,
+                                          **request)
+                query_ids = response['QueryExecutionIds']
+                next_token = response.get('NextToken')
```

Comment:
I prefer to write explicit default values.
`next_token = response.get('NextToken', None)`